src/shared/generator.c

```diff
@@ -694,6 +695,91 @@ int generator_hook_up_pcrfs(
         return generator_add_symlink_full(dir, where_unit, "wants", pcrfs_unit_path, instance);
 }
 
+int generator_hook_up_quotacheck(
+                const char *dir,
+                const char *what,
+                const char *where,
+                const char *target,
+                const char *fstype) {
+
+        const char *quotacheck_unit, *quotacheck_unit_path;
+        _cleanup_free_ char *where_unit = NULL, *instance = NULL;
+        int r;
+
+        assert(dir);
+        assert(where);
+
+        if (!isempty(fstype) && !streq(fstype, "auto")) {
+                if (!fstype_needs_quota(fstype))
+                        return log_warning_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Quota was requested for %s, but not supported, ignoring: %s", what, fstype);
+        } else
+                return log_warning_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Couldn't determine filesystem type for %s, Quota cannot be activated", what);
+
+
+        r = unit_name_from_path(where, ".mount", &where_unit);
+        if (r < 0)
+                return log_error_errno(r, "Failed to make unit name from path '%s': %m", where);
+
+        /* quotacheck unit for system root */
+        if (path_equal(where, "/")) {
+                quotacheck_unit = SPECIAL_QUOTACHECK_ROOT_SERVICE;
+                return generator_add_symlink(dir, where_unit, "wants", quotacheck_unit);
```

Comment:
Why not simply pass `SPECIAL_QUOTACHECK_ROOT_SERVICE` to `generator_add_symlink()`?